### ISSUE TITLE
Unable to use numeric enums as keys of a map

### ISSUE THREAD

**Issue by nevir** [Microsoft/TypeScript#16842]
This appears to be a regression of #2491 (& #12425?)

**TypeScript Version:**  2.4.1

**Code**

Expected that this snippet does not produce errors, but it does:

```ts
enum Things { ONE, TWO }
// Error: Type 'Things' is not assignable to type 'string'. 
type ThingMap = {[TKey in Things]: boolean};
```
[[Playground](https://www.typescriptlang.org/play/index.html#src=enum%20Things%20%7B%20ONE%2C%20TWO%20%7D%0A%2F%2F%20Error%3A%20Type%20'Things'%20is%20not%20assignable%20to%20type%20'string'.%20%0Atype%20ThingMap%20%3D%20%7B%5BTKey%20in%20Things%5D%3A%20boolean%7D%3B%0A)]

**Comment by RyanCavanaugh** [comment]
What version did this work in? As far as I can tell this has never been legal
```
=== latest ===

clipboard.ts:3 type ThingMap = {[TKey in Things]: boolean}
                                         ~~~~~~
  TS2322: Type 'Things' is not assignable to type 'string'.
=== 2.3 ===

clipboard.ts:3 type ThingMap = {[TKey in Things]: boolean}
                                         ~~~~~~
  TS2322: Type 'Things' is not assignable to type 'string'.
=== 2.2 ===

clipboard.ts:3 type ThingMap = {[TKey in Things]: boolean}
                                         ~~~~~~
  TS2322: Type 'Things' is not assignable to type 'string'.
=== 2.1 ===

clipboard.ts:3 type ThingMap = {[TKey in Things]: boolean}
                                         ~~~~~~
  TS2322: Type 'Things' is not assignable to type 'string'.
=== 2.0 ===

clipboard.ts:3 type ThingMap = {[TKey in Things]: boolean}
                                ~~~~~~~~~~~~~~~~
  TS1170: A computed property name in a type literal must directly refer to a built-in symbol.

clipboard.ts:3 type ThingMap = {[TKey in Things]: boolean}
                                ~~~~~~~~~~~~~~~~
  TS2464: A computed property name must be of type 'string', 'number', 'symbol', or 'any'.

clipboard.ts:3 type ThingMap = {[TKey in Things]: boolean}
                                 ~~~~
  TS2304: Cannot find name 'TKey'.
=== 1.8 ===

clipboard.ts:3 type ThingMap = {[TKey in Things]: boolean}
                                ~~~~~~~~~~~~~~~~
  TS1170: A computed property name in a type literal must directly refer to a built-in symbol.
```

**Comment by nevir** [comment]
@RyanCavanaugh doh, sorry! I misread https://github.com/Microsoft/TypeScript/issues/2491#issuecomment-267914140 (and the followup comment) as if it was supported

I think we can probably close this as a dupe of https://github.com/Microsoft/TypeScript/issues/13042

**Comment by amitbeck** [comment]
@RyanCavanaugh but is it ever going to be supported/legal?

**Comment by nevir** [comment]
@amitbeck it looks like it's on the roadmap: https://github.com/Microsoft/TypeScript/issues/13042#issuecomment-274659226

**Comment by amitbeck** [comment]
@nevir I know, I've commented on that exact issue :) Didn't see the milestone change, thanks for reminding me.
